### ISSUE TITLE
Allow to import multiple owner keys for signing

### ISSUE THREAD

**Issue by tschubotz** [gnosis/safe#387]
# What is this feature about? (1 sentence)
This allows a user to import more than 1 private key on 1 device.

# Why is it needed? What is the value? For whom do we build it?

- This is related to the problem "Only possible to import one private key on mobile" that we have on our roadmap.
- It's a good practice to use a different owner for every Safe you own.
- Currently, we restricted importing owners keys to 1.
- ~80% of our users have more than 1 Safe added ([Source](https://datastudio.google.com/u/0/reporting/f1538189-3dbb-4a25-82d5-3f9f0d8a1927/page/ub9tB))

# High-level overview of the feature

- User can import multiple private keys (via seed or directly the pk) one after each other.
- User can use any imported key for signing
- User can remove keys one by one.
- User can rename any key.

# Spec
[Multiple Owner Keys for Signing](https://docs.google.com/document/d/13GirjSwieQaqUPz-8HXzJP_rfgmobSp7E4TpbnLlb0M/edit#
)
# Misc

- [Kick off doc](https://docs.google.com/document/d/1Nfi0ycN3v1WIO0TkJOfhxbdVoNfuNvOuc8hht0XCBj4/edit# )

**Comment by tschubotz** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [Keychain (allow to import multiple keys)](https://trello.com/c/aNnrs69S/81-keychain-allow-to-import-multiple-keys)

**Comment by posthnikova** [comment]
Wireframes with basic user flows.   

**1) Import**
Additional steps in existing flow – add owner key name, create passcode and enable biometrics. Passcode step can be skipped. The amount of keys is shown in the menu https://prnt.sc/yyk0pe

![image](https://user-images.githubusercontent.com/45787536/107516851-466d1400-6bad-11eb-8f62-c0e97b6f375d.png)

When a key was already imported we can disable it in the list and show custom name:

![image](https://user-images.githubusercontent.com/45787536/107516907-5a187a80-6bad-11eb-8366-d195b6c43a46.png)

**2) Edit name**
There is a button to edit name.

![image](https://user-images.githubusercontent.com/45787536/107516963-6dc3e100-6bad-11eb-940d-88897cb5dd06.png)

**3) Remove**
Removing is accessed from the list and on the name screen.

![image](https://user-images.githubusercontent.com/45787536/107516998-79afa300-6bad-11eb-82f6-403169e32058.png)

**4) Sign**
Choosing the key with which to sign happens in a bottom sheet. When there is only one eligible key the bottom sheet is not shown. Open question: what if the passcode was not set? Then there's no additional step after you tap 'Confirm'.

![image](https://user-images.githubusercontent.com/45787536/107517035-87652880-6bad-11eb-9ae5-808d4739162a.png)

**Comment by KristinaMayman** [comment]
What is the transition going to be between the screen with the snack bar saying that the key was successfully imported and the one where the user need to set a passcode? Is it just going to be timed? I feel like that might be a bit surprising.

**Comment by posthnikova** [comment]
@KristinaMayman Yes, it's going to be surprising, you think you're done and then something else pops up. Maybe stay in the import flow?

![image](https://user-images.githubusercontent.com/45787536/107521413-ce095180-6bb2-11eb-8615-f0a92253cc11.png)

**Comment by biafra23** [comment]
![image](https://user-images.githubusercontent.com/246473/107521980-791a0b00-6bb3-11eb-870b-550a5db9ce42.png)

I'd rephrase this to: 
```
The passcode will be needed to open the app and sign transactions.
```

**Comment by biafra23** [comment]
![image](https://user-images.githubusercontent.com/246473/107522331-e5950a00-6bb3-11eb-8cb6-0adaeb7d214d.png)

I am not sure, but should it be singular for party? 
```
[...]will not be shared with Gnosis or any other third party
```

A quick check reveals. Both versions are common.

**Comment by biafra23** [comment]
![image](https://user-images.githubusercontent.com/246473/107524208-d31bd000-6bb5-11eb-8ee9-51e9ba561dd5.png)

It's not wrong (IMHO). But maybe one of the following is better:
```
Chose the key to use for signing the transaction
```
or
```
Chose the key to use to sign the transaction
```
or (Not sure this is correct:)
```
Chose the key to sign the transaction
```

**Comment by tschubotz** [comment]
I like the flows! 

Regarding number 4:

- After signing with one of the keys, I feel like the user should still be able to sign with the other one afterwards, no?

![image](https://user-images.githubusercontent.com/1380529/107525093-c8156f80-6bb6-11eb-8795-2639b41753f7.png)

Shouldn't this rather say: "Choose the _account_ to sign the transaction"?

**Comment by DmitryBespalov** [comment]
> I like the flows!
> 
> Regarding number 4:
> 
> * After signing with one of the keys, I feel like the user should still be able to sign with the other one afterwards, no?
> 
> ![image](https://user-images.githubusercontent.com/1380529/107525093-c8156f80-6bb6-11eb-8795-2639b41753f7.png)
> 
> Shouldn't this rather say: "Choose the _account_ to sign the transaction"?

Aren't we using the "Owner key" everywhere in the app? Also, we use "Confirm" not sign, or maybe I'm confusing something

**Comment by posthnikova** [comment]
**Zeplin:**   
Android: https://zpl.io/bJXwABJ
iOS: https://zpl.io/b6Xe70y
Dark mode assets for iOS: https://zpl.io/2G8DdXm, https://zpl.io/2vERkvP   

@biafra23 Your comments have been implemented: https://zpl.io/a3L8jRY, https://zpl.io/VYe4pXM. Does that work for you?

> Shouldn't this rather say: "Choose the account to sign the transaction"?

@tschubotz Agree with Dima, we should say 'Owner key' like everywhere. 

![image](https://user-images.githubusercontent.com/45787536/108184776-df4ee280-710b-11eb-9439-08353c4cf1c2.png)

**Comment by liliya-soroka** [comment]
Verified for ios app - 2.12.0 (334) with a note: the issues with "requires your confirmation" notifications exist. The changes on the back-end side are required to fix them

**Comment by liliya-soroka** [comment]
Verified
